firebase-segmentation/src/main/java/com/google/firebase/segmentation/FirebaseSegmentation.java

```diff
@@ -191,50 +215,67 @@ public static FirebaseSegmentation getInstance(@NonNull FirebaseApp app) {
    * </pre>
    */
   private Task<Void> clearCustomInstallationId() {
-    Task<InstanceIdResult> instanceIdResultTask = firebaseInstanceId.getInstanceId();
-    Task<Boolean> firstUpdateCacheResultTask =
-        instanceIdResultTask.onSuccessTask(
-            instanceIdResult ->
-                localCache.insertOrUpdateCacheEntry(
-                    CustomInstallationIdCacheEntryValue.create(
-                        "",
-                        instanceIdResult.getId(),
-                        CustomInstallationIdCache.CacheStatus.PENDING_CLEAR)));
-
-    Task<Code> backendRequestResultTask =
-        firstUpdateCacheResultTask.onSuccessTask(
-            firstUpdateCacheResult -> {
-              if (firstUpdateCacheResult) {
-                String iid = instanceIdResultTask.getResult().getId();
-                String iidToken = instanceIdResultTask.getResult().getToken();
-                return backendServiceClient.clearCustomInstallationId(
-                    Utils.getProjectNumberFromAppId(firebaseApp.getOptions().getApplicationId()),
-                    iid,
-                    iidToken);
-              } else {
-                throw new SetCustomInstallationIdException(
-                    "Failed to update client side cache", Status.CLIENT_ERROR);
-              }
-            });
-
-    Task<Boolean> finalUpdateCacheResultTask =
-        backendRequestResultTask.onSuccessTask(
-            backendRequestResult -> {
-              if (backendRequestResult == Code.OK) {
-                return localCache.clear();
-              } else {
-                throw new SetCustomInstallationIdException(Status.BACKEND_ERROR);
-              }
-            });
-
-    return finalUpdateCacheResultTask.onSuccessTask(
-        finalUpdateCacheResult -> {
+
+    TaskCompletionSource<Void> taskCompletionSource = new TaskCompletionSource<>();
+
+    executor.execute(
+        () -> {
+          InstanceIdResult instanceIdResult;
+          try {
+            instanceIdResult = Tasks.await(firebaseInstanceId.getInstanceId());
+          } catch (Exception e) {
```

Comment:
Is there any particular exception you are anticipating here? as catching `Exception` is very broad: http://go/java-practices/exceptions#catch_exception